### PR TITLE
add a button to demo to show embedding

### PR DESCRIPTION
**Pull request by paulfitz**
Adds a button to the code snippet in the demo to actually run it and replace the code with the spreadsheet it describes.

A bit lame because there's no way to undo other than reload :-), but, baby-steps.

[Screencast from 07-27-2023 07:49:28 AM.webm](https://github.com/gristlabs/grist-static/assets/118367/e77c4c3b-8528-4ebb-ab03-b427943ce077)

Also updates to latest `grist-core` to pick up https://github.com/gristlabs/grist-core/pull/596

### CODE REVIEW

**Comment by paulfitz** [comment]
Added an example of loading CSV as well. These examples are handy for quickly testing on different browsers and devices.

**Comment by paulfitz** [comment]
Also swapped out meme example with flashcards (more useful).

File: docs/index.html [hunk 1]
```diff
@@ -70,16 +64,81 @@ <h3>Example 3</h3>
   &lt;body&gt;
     &lt;script&gt;
       bootstrapGrist({
-        initialFile: 'investment-research.grist',   // path to .grist
-        name: 'Investment Research',                // optional name
+        initialFile: 'weresheet.grist',   // optional path to .grist
```
Comment (alexmojaki) [inline docs/index.html:67]: Why the change in this example?

File: docs/index.html [hunk 2]
```diff
@@ -70,16 +64,81 @@ <h3>Example 3</h3>
...
+        name: 'Werewolf Fact Sheet',      // optional name
+        elementId: 'were-div',            // optional element name
+        light: true                       // optional minimalist style
```
Comment (alexmojaki) [inline docs/index.html]: I'm noticing now that `light: true` doesn't feel very readable. It sounds like it's turning off dark mode. `embed: true` would match what actually happens in the code and seems slightly more understandable but not great. Maybe `minimalUI: true`?

File: docs/index.html [hunk 3]
```diff
@@ -70,16 +64,81 @@ <h3>Example 3</h3>
...
+      light: true
+    });
+    document.getElementById('were-click').style.display = 'none';
+    return false;
+  });
+  document.getElementById('csv-click').addEventListener('click', function() {
+    const div = document.getElementById('csv-div');
+    div.style.height = '500px';
+    div.style.padding = '0';
+    div.style.border = '1px solid gray';
```
Comment (alexmojaki) [inline docs/index.html:131]: Why is this different from the were-click button?

File: docs/index.html (hunk 2 above)
Comment (paulfitz) [reply docs/index.html]: I agree the name isn't great. It is usable without embedding (i.e. when Grist takes over whole page) which is why I avoided `embed`, which I would have preferred otherwise since it is a query parameter we have documented https://support.getgrist.com/FAQ/#can-i-embed-grist-into-my-website and it would have been nice to "rhyme" with it.

`minimalUI` would be ok. How would you feel about `thin` or `lightweight` or (consults chatbot) `svelte`, `ethereal`, `dainty`  ... ? Or we could focus on the "single page" aspect of the bundle of changes with `singlePage`?

File: docs/index.html (hunk 1 above)
Comment (paulfitz) [reply docs/index.html:67]: This was moving the weresheet live example from its current location as a bulletpoint.

File: docs/index.html (hunk 3 above)
Comment (paulfitz) [reply docs/index.html:131]: The embeds look better without an extra border, but this one is very slow to start, and a border makes it work better with the "watch this space" directive - and gives a little sense that something may be coming eventually.

File: docs/index.html (hunk 2 above)
Comment (alexmojaki) [reply docs/index.html]: I think I like `singlePage` best.

I'm actually confused. I thought that with this code:

```
  if (gristOverrides.light) {
    location.searchParams.set('embed', 'true');
    location.searchParams.set('style', 'light');
  }
```

Setting `embed` would make the doc read only, but it's not.

**Comment by alexmojaki** [comment]
Overall I think what's in this PR is fine, and changing the `light` param is not really in scope, but at the same time it would be ideal to avoid documenting something we want to change.

File: docs/index.html (hunk 2 above)
Comment (paulfitz) [reply docs/index.html]: Renamed to `singlePage` (though PR isn't updating right now)
Screenshot from 2023-07-28 11-20-48

Yes the `embed` thing is a bit funky.

Comment (alexmojaki) [reply docs/index.html]: Do you know why it doesn't make things read-only?

Comment (paulfitz) [reply docs/index.html]: I was afraid you were going to ask that :). I'll investigate.

Comment (paulfitz) [reply docs/index.html]: Turns out it is because the doc technically qualifies as a fork currently, and the read-only logic has an exception for forks: https://github.com/gristlabs/grist-core/blob/3d84014116a3f07010131ef9989073e763e3031d/app/common/gristUrls.ts#L399-L400

If we change the doc setup later (I expect we will, the relevant code is very hacky), then the read-only mode could kick in - but it'd be easy to fix by specifying an explicit mode.

**Review by alexmojaki — APPROVED** [review]
